src/schedule/dma_multi_chan_domain.c

```diff
@@ -280,61 +284,63 @@ static bool dma_multi_chan_domain_is_pending(struct ll_schedule_domain *domain,
 	int j;
 
 	for (i = 0; i < dma_domain->num_dma; ++i) {
-		for (j = 0; j < dmas[i].plat_data.channels; ++j) {
-			if (!*comp) {
-				status = dma_interrupt_legacy(&dmas[i].chan[j],
-							      DMA_IRQ_STATUS_GET);
-				if (!status)
+		if (dmas[i].chan != NULL) {
```

Comment:
Ditto here. I assume code on L288-342 is not modified at all, you just want to skip the NULL channels

Reply:
done